libkbfs/disk_block_cache.go

```diff
@@ -409,14 +411,20 @@ func (cache *DiskBlockCacheStandard) Get(ctx context.Context, tlfID tlf.ID,
 	blockKey := blockID.Bytes()
 	entry, err := cache.blockDb.Get(blockKey, nil)
 	if err != nil {
-		return nil, kbfscrypto.BlockCryptKeyServerHalf{},
+		return nil, kbfscrypto.BlockCryptKeyServerHalf{}, false,
 			NoSuchBlockError{blockID}
 	}
-	err = cache.updateMetadataLocked(ctx, tlfID, blockKey, len(entry))
+	md, err := cache.getMetadata(blockID)
 	if err != nil {
-		return nil, kbfscrypto.BlockCryptKeyServerHalf{}, err
+		return nil, kbfscrypto.BlockCryptKeyServerHalf{}, false, err
+	}
+	err = cache.updateMetadataLocked(ctx, tlfID, blockKey, len(entry),
+		hasPrefetched)
```

Comment:
Should this be `md.HasPrefetched`?  Otherwise you'll just overwrite it with `false` on every Get.